### DISCUSSION TITLE
Is there anyway i can add password protection when generating pdf

### DISCUSSION THREAD

**Discussion by bhoopaljanga** [QuestPDF/QuestPDF#263] [Q&A]
I am trying to generate password protected pdf, Is there anyway i can add password protection when generating pdf

**Comment by MarcinZiabek** [comment]
Hi 😁 I am afraid that the password protection feature is not available at this moment.

**Reply by bhoopaljanga** [reply]
ok any links/suggestions on how we can do it

**Reply by bennetbo** [reply]
This should be pretty straightforward with PdfSharp.
[See this link.](http://www.pdfsharp.net/wiki/ProtectDocument-sample.ashx)

You can use QuestPDF to produce the Pdf file and then use PdfSharp to open the existing Pdf and protect it with a password.

Edit: [Pdf Sharp Repo](https://github.com/empira/PDFsharp)

**Reply by douglott** [reply]
I'm doing the same.  Please let me know if you need the code.  Marcin - ❤️ QuestPdf!

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response.

We're happy to share that PDF password protection and encryption are now natively supported starting from version 2024.12.0. You can encrypt documents with 40-bit, 128-bit, or 256-bit encryption and configure granular permissions. You can find the documentation and usage examples [here](https://www.questpdf.com/concepts/document-operations.html)

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
